harpoon/steps.go

```diff
@@ -24,6 +30,56 @@ func RegisterStep(expression string, step interface{}) {
 
 func getSteps(ctx *godog.ScenarioContext) {
 	for _, step := range registeredSteps {
-		ctx.Step(step.expression, step.step)
+		ctx.Step(step.expression, injectTestingT(internaltesting.WrapWithPanicHandler(false, internaltesting.ExitBehaviorNone, step.step)))
+	}
+}
+
+type empty struct{}
+
+func injectTestingT(fn interface{}) interface{} {
+	fnValue := reflect.ValueOf(fn)
+	fnType := fnValue.Type()
+	inTypes := []reflect.Type{}
+	for i := 0; i < fnType.NumIn(); i++ {
+		inTypes = append(inTypes, fnType.In(i))
+	}
+	outTypes := []reflect.Type{}
+	for i := 0; i < fnType.NumOut(); i++ {
+		outTypes = append(outTypes, fnType.Out(i))
+	}
+
+	path := reflect.TypeOf(empty{}).PkgPath()
+
+	// handle the case of ctx.Context, TestingT...
+	if len(inTypes) > 1 && isType("context", "Context", inTypes[0]) && isType(path, "TestingT", inTypes[1]) {
```

Comment:
If you're okay with importing the types you're testing for you can check with `inTypes[0] == reflect.TypeFor[context.Context]`

Oh wait you're using that below. Any reason not to use it here? If path is this package path, can you not reference TestingT directly?

Reply:
Yeah, I wrote the code below after I wrote this and forgot to change things to use `TypeFor` 😅